### ISSUE TITLE
Implement chat messaging

### ISSUE THREAD

**Issue by Titogelo** [WELTEN/dojo-ibl-mobile#14]
For user to able to exchange messages

**Comment by rafaelklaessen** [comment]
How are we going to request messages? There is no resumption token mechanism, so do we request them all at once?

**Comment by rafaelklaessen** [comment]
OR, we can use recentfirst, and then just request the first 20 messages... If the user scrolls up, we'd request the first 40 messages, 60 messages, 80 messages etc

**Comment by rafaelklaessen** [comment]
We could determine if we have all messages or not by using the length of the messages array in the response. `if (the received amount of messages < the amount we specified)`, we have them all and the user can't request more.

**Comment by rafaelklaessen** [comment]
The point however is, is that we don't request 20 more every time. We request all the previous messages + 20. So, if the user just wants the first 20 messages, we save server power. If the user wants the first 80, we waste server power because we actually request 20+40+60+80 = 200 messages.

**Comment by rafaelklaessen** [comment]
So, what do we do? We could also request in bigger blocks (blocks of 100 messages or something) as users probably won't scroll that much up.

**Comment by Titogelo** [comment]
I think we have resumptionToken mechanism, check this request URL and don't forget the **from** parameter! 

http://dojo-ibl.appspot.com/rest/messages/runId/5726267328430080/default?from=0

Additionally, once you have all the messages locally try to work with the server time that comes with each block of messages in order to save **server power**. Using the servertime you only retrieve the new messages from the  last time you checked the server. Of course you need to place the servertime timestamp in the from parameter!

So, to sum up. I suggest you two things:

1) retrieve the first block of messages and give the option to the download more if they scroll up. 
2) if they scroll up to load more messages use the resumptionToken
3) if you already have messages locally (first you need to check that) use the last servertime you retrieved to get the possible new messages since the last time.

Let me know if you have questions :)

**Comment by rafaelklaessen** [comment]
Hm, the API gets the oldest messages first, but we're showing the newest messages first. How are we going to do that?

**Comment by Titogelo** [comment]
Oh okay, that's true. I can fixed that adding a parameter to sort by lastModificationDate desc. Although it's work for 10 minutes I can't do it right now. For now show them as ASC and I will give you a sign when the parameter to show in DESC mode is ready.
